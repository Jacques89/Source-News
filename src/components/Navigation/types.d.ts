type NavigationProps = {
  category?: string
  changeCategory: (a: string) => void
  menuOpen?: boolean
}