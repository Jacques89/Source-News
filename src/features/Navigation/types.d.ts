type NavigationProps = {
  category?: string
  changeCategory: (category: string) => void
  menuOpen?: boolean
}