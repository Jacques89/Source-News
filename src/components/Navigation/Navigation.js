import React, { Component } from 'react'

import { FaBars } from 'react-icons/fa'

import '../../styles/Navigation.scss'

export default class Navigation extends Component {
  state = {
    menuOpen: false
  } 

  toggleMenu = e => {
    let toggle = !this.state.menuOpen

    this.setState({
        menuOpen: toggle
    })
  }

  isMenuOpen = () => {
    if (this.state.menuOpen) {
      return 'block'
    } else {
      return 'none'
    }
  }
  
    

  render() { 
    return (
      <header>
        <h1>Source News</h1>
        <h3>Straight from the sources mouth</h3>
        <FaBars className='icon-menu' onClick={this.toggleMenu} />
        <nav style={{ display: this.isMenuOpen() }}>
          <ul>
            <li><a href="#">Politics</a></li>
            <li><a href="#">Economics</a></li>
            <li><a href="#">Sport</a></li>
            <li><a href="#">Culture</a></li>
            <li><a href="#">Technology</a></li>
          </ul>
        </nav>
      </header>
    )
  }
}