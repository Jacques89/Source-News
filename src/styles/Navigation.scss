header {
  width: 100%;
  height: auto;
  background-color: antiquewhite;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
}
h1 {
  font-size: 3rem;
  margin: 1rem 0;
  font-family: 'Abril Fatface', serif;
  letter-spacing: 0.2rem;
  text-align: center;
  cursor: pointer;
}

.icon-menu {
  font-size: 2rem;
  cursor: pointer;
  margin: 0.5rem 0;
  z-index: 90;

  &:hover {
      color: rgb(77, 66, 49);
  }
}

nav {
  width: 100%;
  height: calc(100vh - 145px);
  display: none;
}

ul {
  width: 100%;
  height: calc(100% - 145px);
  display: flex;
  flex-direction: column;
  background-color: antiquewhite;
  position: absolute;
  top: 14.5rem;

  li {
    width: 100%;
    height: 100%;
    font-size: 1.5rem;
    text-align: center;
    padding: 2rem, 0.5rem;
    color: rgb(7, 6, 6);
    cursor: pointer;
    display: flex;

    a {
      align-items: center;
      justify-content: space-around;
    }

    &:hover {
      background-color: rgb(77, 66, 49);
      color: #fff;
      font-weight: 600;
    }
  }
}

@media screen and (min-width: 670px) {
  h1 {
    font-size: 5rem;
  }

  .icon-menu {
    display: none;
  }

  nav {
    display: block !important;
    z-index: 0 !important;
    height: auto;
  }

  ul {
    width: 100%;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-around;
    background-color: transparent;
    position: relative;
    top: 0;

    li {
      width: 100%;
      height: 100%;
      font-size: 1.5rem;
      text-align: center;
      padding: 0.5rem;

      a {
        color: #000;
      }

      &:hover {
        background-color: rgb(77, 66, 49);
        color: #fff;
      }
    }
  }
}

@media screen and (max-width: 400px){
  .icon-menu {
    margin: 0.5rem;
    align-self: flex-end;
  }
}