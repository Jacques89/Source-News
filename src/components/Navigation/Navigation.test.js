import React from 'react'
import Navigation from './Navigation'

import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

describe('NavigationBar', () => {
  const changeCategory = jest.fn()

  beforeEach(() => {
    render(<Navigation changeCategory={changeCategory} />)
  })

  it('renders a title and sub-title', () => {
    expect(screen.getByRole('heading', {name: /Source News/i }))
    expect(screen.getByRole('heading', {name: /Straight from the sources mouth!/i }))
  })

  it('renders news categories', () => {
    expect(screen.getByTestId('news-categories')).toMatchSnapshot()
  })

  it('changes category when a category is clicked', () => {
    const handleClick = changeCategory
    render(<li onClick={handleClick}>Business</li>)
    userEvent.click(screen.getByTestId('business'))
    expect(handleClick).toHaveBeenCalledTimes(1)
  })
})