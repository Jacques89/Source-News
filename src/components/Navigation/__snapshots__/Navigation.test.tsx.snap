// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar renders correctly 1`] = `
<div
  data-testid="navigation"
  style="margin-top: -1000px; transition-duration: .8s;"
>
  <header>
    <h1>
      Source News
    </h1>
    <h3>
      Straight from the sources mouth!
    </h3>
    <svg
      class="icon-menu"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
      />
    </svg>
  </header>
  <nav
    style="opacity: 0; display: none; z-index: 1;"
  >
    <ul
      aria-label="news-categories"
    >
      <li
        aria-label="category"
        style="background-color: rgb(0, 0, 0); color: rgb(255, 255, 255);"
      >
        World
      </li>
      <li
        aria-label="category"
        style="background-color: rgb(0, 0, 0); color: rgb(255, 255, 255);"
      >
        Business
      </li>
      <li
        aria-label="category"
        style="background-color: rgb(0, 0, 0); color: rgb(255, 255, 255);"
      >
        Sports
      </li>
      <li
        aria-label="category"
        style="background-color: rgb(0, 0, 0); color: rgb(255, 255, 255);"
      >
        Entertainment
      </li>
      <li
        aria-label="category"
        style="background-color: rgb(0, 0, 0); color: rgb(255, 255, 255);"
      >
        Technology
      </li>
    </ul>
  </nav>
</div>
`;
